src/main/scala/io/iohk/ethereum/network/PeerManagerActor.scala

```diff
@@ -68,16 +70,17 @@ class PeerManagerActor(
     peer
   }
 
+  def peerFactory(context: ActorContext, address: InetSocketAddress): ActorRef
 }
 
 object PeerManagerActor {
   def props(nodeStatusHolder: Agent[NodeStatus], storage: Blockchain): Props =
-    Props(new PeerManagerActor(nodeStatusHolder, peerFactory(nodeStatusHolder, storage)))
-
-  def peerFactory(nodeStatusHolder: Agent[NodeStatus], storage: Blockchain): (ActorContext, InetSocketAddress) => ActorRef = { (ctx, addr) =>
-    val id = addr.toString.filterNot(_ == '/')
-    ctx.actorOf(PeerActor.props(nodeStatusHolder, storage), id)
-  }
+    Props(new PeerManagerActor(nodeStatusHolder) {
```

Comment:
to be able to construct peer and pass peer manager to it, will change peer actor